cartographer/mapping/internal/2d/scan_matching/interpolated_tsdf_2d.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_MAPPING_INTERNAL_2D_SCAN_MATCHING_TSDF_INTERPOLATED_TSDF_2D_H_
+#define CARTOGRAPHER_MAPPING_INTERNAL_2D_SCAN_MATCHING_TSDF_INTERPOLATED_TSDF_2D_H_
+
+#include <cmath>
+
+#include "cartographer/mapping/2d/tsdf_2d.h"
+
+namespace cartographer {
+namespace mapping {
+namespace scan_matching {
+
+// Interpolates between TSDF2D pixels with bilinear interpolation.
+//
+// This class is templated to work with Ceres autodiff.
+class InterpolatedTSDF2D {
+ public:
+  explicit InterpolatedTSDF2D(const TSDF2D& grid) : tsdf_(grid) {}
+
+  InterpolatedTSDF2D(const InterpolatedTSDF2D&) = delete;
+  InterpolatedTSDF2D& operator=(const InterpolatedTSDF2D&) = delete;
+
+  // Returns the interpolated correspondence cost at (x,y). Cells with at least
+  // one 'unknown' interpolation point result in "MaxCorrespondenceCost()" with
+  // zero gradient.
+  template <typename T>
+  T GetCorrespondenceCost(const T& x, const T& y) const {
```

Comment:
Since this is x and y separate, I would vote for passing by non-const value

Reply:
For `ceres::Jet` reference is more efficient tahn non-const value.